setup.sh

```diff
@@ -1,35 +1,50 @@
 #!/bin/bash
 set -x
 
-# Juju and Dependencies
+# This script installs juju version 2 and all the dependencies.
+
+if [ -z "$JUJU_USER" ]; then
+  JUJU_USER=ubuntu
+fi
+
+# Refresh the potentially stale apt cache.
 apt-get update -qq
-apt-get install -qy software-properties-common
-apt-add-repository -y ppa:juju/devel
+# Install software tools such as add-apt-repository.
+apt-get install -y software-properties-common
+# devel ppa is required for juju2 beta release
+apt-add-repository -u -y ppa:juju/devel
 # stable ppa is required for charm-tools
-apt-add-repository -y ppa:juju/stable
-apt-get update -qq
+apt-add-repository -u -y ppa:juju/stable
```

Comment:
That can be added in the downstream box image that requires this. It makes more sense to encapsulate that there rather htan here in a lower layer, where its inclusion is effectively hidden unless there are other reasons for having both PPA's enabled in jujubox.